pkg/kv-cache/kv-block/in-memory.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2025 The llm-d Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kvblock
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/llm-d/llm-d-kv-cache-manager/pkg/utils"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultInMemoryIndexSize = 1e8 // TODO: change to memory-size based configuration
+	defaultPodCacheSize      = 10  // number of pods per key
```

Comment:
nit: name can be a bit misleading (compared with comment) since it can be interpreted as the default cache size per Pod? Maybe `defaultPodsPerKey`?